net/dnsfallback/dnsfallback.go

```diff
@@ -250,10 +250,11 @@ func SetCachePath(path string) {
 // logfunc stores the logging function to use for this package.
 var logfunc syncs.AtomicValue[logger.Logf]
 
-// SetLogger sets the logging function that this package will use. The default
-// logger if this function is not called is 'log.Printf'.
-func SetLogger(log logger.Logf) {
-	logfunc.Store(log)
+// SetLogger sets the logging function that this package will use, and returns
+// the old value (which may be nil). The default logger if this function is not
```

Comment:
The default value of the logger is log.Printf

Reply:
Yep, but the default value of the `logfunc` is nil; the top-level `logf` function will fall back to `log.Printf` if so.

Reply:
The value defaults to log.Printf if not explicitly set?